CONTRIBUTING.md

```diff
@@ -113,6 +113,31 @@ If you'd like to contribute example `.bicep` files that showcase abilities of th
 
 **Note:** If you have never submitted a Pull Request or used git before, reading through the [Git tutorial](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/git-tutorial.md) in the azure-quickstart-template repo is a good place to start.
 
+### Snippets
+
+If you'd like to contribute to the collection of snippets:  
+
+* Add a Bicep file here: [`./src/Bicep.LangServer/Snippets/Templates`](./src/Bicep.LangServer/Snippets/Templates)
+  * The file name without extension will be used as the label.
+  * A single line comment at the top of the file will be used as the description.
+  * E.g. [`res-aks-cluster.bicep`](./src/Bicep.LangServer/Snippets/Templates/res-aks-cluster.bicep) results in the following label and description:
+ ![image](https://user-images.githubusercontent.com/6855361/114454607-69617780-9ba0-11eb-80b0-1f9f2d76ae6c.png)
+
+* Add a new folder in the following directory, for an integration test that validates snippet completion: [`./src/Bicep.LangServer.IntegrationTests/Completions/SnippetTemplates`](./src/Bicep.LangServer.IntegrationTests/Completions/SnippetTemplates)
+  * The folder name should match the snippet label/prefix.
+
+* Add a file named main.json
+  * This should include information about placeholder replacements. The test will replace the snippet placeholders and ensure there are no errors.
+
+  * E.g. [`res-aks-cluster/main.bicep`](./src/Bicep.LangServer.IntegrationTests/Completions/SnippetTemplates/res-aks-cluster/main.bicep)
+
+* Add a file named diagnostics.json - this should contain information about expected errors. There should not be any for new contributions, this is used for things like deprecated features.
```

Comment:
@bhsubra - does this make sense? Do you have anything to add to make this clearer?

Reply:
LGTM! Thanks!